### PR TITLE
fix: Don't generate telemetry-random-id in current dir

### PR DESCRIPTION
**Pull request by erezrokah**
🎉 Thank you for making CloudQuery awesome by submitting a PR 🎉

#### Summary

Fixes an issue that on each run of `cloudquery` a telemetry id file is created in the current dir

---

Use the following steps to ensure your PR is ready to be reviewed

- [x] Read the [contribution guidelines](../blob/main/.github/CONTRIBUTING.md) 🧑‍🎓
- [x] Run `go fmt` to format your code 🖊
- [x] Lint your changes via `golangci-lint run --new-from-rev main` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [x] Update or add tests 🧪
- [x] Ensure the status checks below are successful ✅

### CODE REVIEW

File: internal/analytics/client.go [hunk 1]
```diff
@@ -28,8 +28,8 @@ type VersionInfo struct {
 	CommitId  string `json:"commit_id,omitempty"`
 }
 
-// currentHub is the initial Hub with no Client bound and an empty Scope.
-var currentHub = New()
+// Consumers must call analytics.Init before using this package
+var currentHub *Client = nil
```
Comment (erezrokah) [inline internal/analytics/client.go:32]: Not sure this is the right fix.
We could also call `Init()` before each method call in this package if `currentHub = nil` (and maybe add a lock to make it safe)

Comment (roneli) [reply internal/analytics/client.go:32]: Ye that is fine, you can actually do in initAnalytics in cmd package to set the default hub or even wrap it in a sync.Once

Comment (disq) [reply internal/analytics/client.go:32]:
```suggestion
var currentHub *Client
```

**Review by disq — APPROVED** [review]
LGTM with nit/styling

File: internal/analytics/client.go (hunk 1 above)
Comment (erezrokah) [reply internal/analytics/client.go:32]: I'm probably missing something here. We already call `initAnalytics` here https://github.com/cloudquery/cloudquery/blob/811855475622955dcfb323298292bde958f4372d/cmd/root.go#L157.

Where should I add another call?

Comment (disq) [reply internal/analytics/client.go]: I think this is fine

**Pull request merged**